src/FSharp.Stats/Testing/UTest.fs

```diff
@@ -0,0 +1,51 @@
+﻿namespace FSharp.Stats.Testing
+
+// taken/implemented from: https://en.wikipedia.org/wiki/Mann%E2%80%93Whitney_U_test#U_statistic
+module UTest =
+
+    open FSharp.Stats
+    open FSharp.Stats.Testing
+
+    // TO DO:   Bergmann et al. (2000) showed that there are different implementations of this test that lead to different results.
+    //          They implied that some of them are using a false algorithm. Check if the mathematical derivation from above is wrong too.
+    //          Read: https://www.jstor.org/stable/2685616
+    let inline private compute (seq1 : seq<'T>) (seq2 : seq<'T>) =
+        let sortedMerge = 
+            (seq1 |> Seq.map (fun v -> float v, 0), seq2 |> Seq.map (fun v -> float v, 1)) // 0 = first group; 1 = second group
+            ||> Seq.append
+            |> Seq.sortByDescending (fun (v,groupIndex) -> v)
```

Comment:
Is it necessary that the sequences are sorted? The ranking is order-independent and as I can see there is no need for sorting.